datumaro/plugins/data_formats/datumaro_binary/exporter.py

```diff
@@ -4,26 +4,103 @@
 
 # pylint: disable=no-self-use
 
-from io import TextIOWrapper
+import os.path as osp
+import struct
+from io import BufferedWriter
+from typing import Any, Optional
 
+from datumaro.components.dataset_base import IDataset
+from datumaro.components.exporter import ExportContext
 from datumaro.plugins.data_formats.datumaro.exporter import DatumaroExporter
 from datumaro.plugins.data_formats.datumaro.exporter import _SubsetWriter as __SubsetWriter
+from datumaro.plugins.data_formats.datumaro_binary.crypter import Crypter
+from datumaro.plugins.data_formats.datumaro_binary.mapper import DictMapper
 
 from .format import DatumaroBinaryPath
 
 
 class _SubsetWriter(__SubsetWriter):
     """"""
 
-    def _sign(self, fp: TextIOWrapper):
-        fp.write(DatumaroBinaryPath.SIGNATURE.encode("utf-8"))
+    def __init__(self, context: IDataset, ann_file: str, encryption_key: Optional[bytes] = None):
+        super().__init__(context, ann_file)
+        self._fp: Optional[BufferedWriter] = None
+        self._crypter = Crypter(encryption_key)
+
+    def _sign(self):
+        self._fp.write(DatumaroBinaryPath.SIGNATURE.encode())
+
+    def _dump_encryption_field(self) -> int:
+        if self._crypter.key is None:
+            msg = b""
+        else:
+            msg = self._crypter.key
+            msg = self._crypter.encrypt(msg)
+
+        return self._fp.write(struct.pack(f"I{len(msg)}s", len(msg), msg))
+
+    def _dump_header(self, header: Any):
+        msg = DictMapper.forward(header)
+
+        if self._crypter.key is not None:
+            msg = self._crypter.encrypt(msg)
+
+        length = struct.pack("I", len(msg))
+        return self._fp.write(length + msg)
+
+    def _dump_info(self):
+        self._dump_header(self.infos)
+
+    def _dump_categories(self):
+        self._dump_header(self.categories)
```

Comment:
don't we need to have a `_dump_media_types`?

Reply:
`media_type` for each item will be stored by `MediaMapper` but there actually is dataset level `media_type`. This will be dumped in the next PR:
- https://github.com/openvinotoolkit/datumaro/pull/830